### PR TITLE
add level change bottom bar

### PR DESCRIPTION
**Pull request by tciesek**
Before I merge that I need to change width and height for first and second floor in `Level` enum (blocked till Sunday - don't merge it please even if it works 😄but you can review, it is small change).

I have also done some small refactorings here and there while I was writing code.

### CODE REVIEW

**Comment by tciesek** [comment]
I have done some small refactorings by the way I was writing the code.

File: app/src/main/java/com/grey/virtualfais/MapViewFragment.java [hunk 1]
```diff
@@ -3,74 +3,92 @@
...
-        tileView.setShouldLoopScale( false );
+        tileView.setShouldLoopScale(false);
+    }
+
+    private void drawPath(TileView tileView, DetectClick detectClick, PathDrawer pathDrawer, int x, int y) {
+        Log.d("HotSpot", "X/Y " + x + " " + y);
+        int scaledX = (int) (x / tileView.getScale());
+        int scaledY = (int) (y / tileView.getScale());
+        Log.d("HotSpot", "Scaled X/Y " + scaledX + " " + scaledY + " Scale: " + tileView.getScale());
+        Room r = detectClick.getClosestRoom(scaledX, scaledY, 0);
```
Comment (grey277) [inline app/src/main/java/com/grey/virtualfais/MapViewFragment.java]: You should change last parameter to currently loaded layer to change mask also.

File: app/src/main/java/com/grey/virtualfais/MainActivity.java [hunk 2]
```diff
@@ -112,9 +117,29 @@ public boolean onNavigationItemSelected(@NonNull MenuItem item) {
...
+
+            performLevelChange(title.charAt(title.length() - 1));
+            return true;
+        });
+
+        attachFragment(fragment, MainFragment.TAG);
     }
 
+    private MainFragment getMainFragment() {
+        MainFragment fragment = MainFragment.newInstance();
```
Comment (grey277) [inline app/src/main/java/com/grey/virtualfais/MainActivity.java]: Is creating new fragment is good idea? Is it not possible to reuse old fragment?

File: app/src/main/java/com/grey/virtualfais/MainActivity.java [hunk 3]
```diff
@@ -112,9 +117,29 @@ public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                     }
                 });
 
-        attachFragment(MainFragment.newInstance(), MainFragment.TAG);
+
+        MainFragment fragment = getMainFragment();
+
+        bottomNavigation.setOnNavigationItemSelectedListener(item -> {
+            CharSequence title = item.getTitle();
```
Comment (grey277) [inline app/src/main/java/com/grey/virtualfais/MainActivity.java]: I think selecting layer number by it's title is not the best idea, but lets left it as it is working.

Comment (tciesek) [reply app/src/main/java/com/grey/virtualfais/MainActivity.java]: Well, it worked :D But I think now is better :)

File: app/src/main/java/com/grey/virtualfais/MapViewFragment.java (hunk 1 above)
Comment (tciesek) [reply app/src/main/java/com/grey/virtualfais/MapViewFragment.java]: I refactored DetectClick instead :P

File: app/src/main/java/com/grey/virtualfais/MainActivity.java (hunk 2 above)
Comment (tciesek) [reply app/src/main/java/com/grey/virtualfais/MainActivity.java]: No it wasn't a good idea. It was why it was so slow but didn't know how to do it better. The solution was to implement `BitmapProvider`

Comment (tciesek) [reply app/src/main/java/com/grey/virtualfais/MainActivity.java]: But this method here is needed to create this view for the first time with level parameter :)

**Comment by grey277** [comment]
Looks very good, changing maps/masks is working and detecting click is also working.

**Pull request merged**